Decred Wallet/Features/TransactionDetails/TransactionDetailsViewController.swift

```diff
@@ -8,262 +8,308 @@ import UIKit
-        self.navigationItem.leftBarButtonItem = UIBarButtonItem(image: UIImage(named: "left-arrow"),
-                                                                style: .done, target: self,
-                                                                action: #selector(back))
-       
-        let optionsMenuButton = UIButton(type: .custom)
-        optionsMenuButton.setImage(UIImage(named: "right-menu"), for: .normal)
-        optionsMenuButton.addTarget(self, action: #selector(showOptionsMenu), for: .touchUpInside)
-        optionsMenuButton.frame = CGRect(x: 0, y: 0, width: 10, height: 51)
-        self.navigationItem.rightBarButtonItems = [
-            UIBarButtonItem(customView: optionsMenuButton)
-        ]
-        
+
+        self.showOrHideDetailsBtn.addBorder(atPosition: .top, color: UIColor.appColors.gray, thickness: 1)
+
         if self.transaction == nil && self.transactionHash != nil {
             let txHash = Data(fromHexEncodedString: self.transactionHash!)!
             var getTxError: NSError?
             let txJsonString = WalletLoader.shared.firstWallet?.getTransaction(txHash, error: &getTxError)
             if getTxError != nil {
                 print("wallet.getTransaction error", getTxError!.localizedDescription)
             }
-            
+
             do {
-                self.transaction = try JSONDecoder().decode(Transaction.self, from:(txJsonString!.utf8Bits))
+                self.transaction = try JSONDecoder().decode(Transaction.self, from: (txJsonString!.utf8Bits))
             } catch let error {
                 print("decode transaction error:", error.localizedDescription)
             }
         }
-        
-        self.prepareTransactionDetails()
+
+        self.prepareGeneralTxDetails()
+        self.displayTitle()
     }
-    
-    fileprivate func prepareTransactionDetails() {
-        var confirmations: Int32 = 0
-        if self.transaction.blockHeight != -1 {
-            confirmations = WalletLoader.shared.firstWallet!.getBestBlock() - Int32(self.transaction.blockHeight) + 1
-        }
-        
-        let isConfirmed = Settings.spendUnconfirmed || confirmations > 1
-        let status = isConfirmed ? LocalizedStrings.confirmed : LocalizedStrings.pending
-        let textColor = isConfirmed ? #colorLiteral(red: 0.2549019608, green: 0.7490196078, blue: 0.3254901961, alpha: 1) : #colorLiteral(red: 0.2392156863, green: 0.3960784314, blue: 0.6117647059, alpha: 1)
-        
-        let txAmount = Utils.getAttributedString(
-            str: "\(self.transaction.dcrAmount.round(8))",
-            siz: 13,
-            TexthexColor: UIColor.appColors.darkBlue
-        )
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        // calculate maximum height of transactionDetailsTable to take up
+        self.transactionDetailsTable.maxHeight = self.view.frame.size.height
+            - self.view.frame.origin.y
```

Comment:
Isn't this always 0? And isn't `self.view.frame.size.height` equal to the screen height? If so, that would imply that this dialog/modal can potentially take up the entire screen which isn't as portrayed in the mockup.

Reply:
Yes, and I removed it. Although `self.view.frame.size.height` is not equal to screen height, it is smaller.